azkaban-common/src/main/java/azkaban/jobExecutor/utils/process/AzkabanProcess.java

```diff
@@ -234,4 +254,28 @@ public String toString() {
     return "Process(cmd = " + Joiner.on(" ").join(cmd) + ", env = " + env
         + ", cwd = " + workingDir + ")";
   }
+
+  public boolean isExecuteAsUser() {
+    return isExecuteAsUser;
+  }
+
+  public void setExecuteAsUser(boolean isExecuteAsUser) {
```

Comment:
better to create a new constructor to take in these params, they should be ready only once the process is initialized.